part2/src/components/CardList.js

```diff
@@ -0,0 +1,75 @@
+import "./CardList.css";
+import noImageBackground from "../assets/card_no_image_background.svg";
+
+function formatDate(value) {
+  const date = new Date(value);
+  return `${date.getFullYear()}. ${date.getMonth() + 1}. ${date.getDate()}.`;
+}
+
+function calculateTime(value) {
+  const MINUTE = 60 * 1000;
+  const HOUR = 60 * MINUTE;
+  const DAY = 24 * HOUR;
+  const MONTH = 31 * DAY;
+  const YEAR = 12 * MONTH;
+
+  const date = new Date() - new Date(value);
+  if (date < HOUR) {
+    const d = parseInt(date / MINUTE);
+    if (d < 2) return "1 minutes ago";
+    return `${d} minutes ago`;
+  } else if (date < DAY) {
+    const d = parseInt(date / HOUR);
+    if (d === 1) return "1 hour ago";
+    return `${d} hours ago`;
+  } else if (date < MONTH) {
+    const d = parseInt(date / DAY);
+    if (d === 1) return "1 day ago";
+    return `${d} days ago`;
+  } else if (date < YEAR) {
+    const d = parseInt(date / MONTH);
+    if (d === 1) return "1 month ago";
+    return `${d} months ago`;
+  } else {
+    const d = parseInt(date / YEAR);
+    if (d === 1) return "1 year ago";
+    return `${d} years ago`;
+  }
+}
```

Comment:
지금도 코드는 너무 잘 작성해 주셨습니다.
하지만 해당 코드를 [early return](https://devscb.tistory.com/130) 을 이용하면 더욱더 이해하기 손쉬운 코드가 될 것 같습니다.

```suggestion
function calculateTime(value) {
  const MINUTE = 60 * 1000;
  const HOUR = 60 * MINUTE;
  const DAY = 24 * HOUR;
  const MONTH = 31 * DAY;
  const YEAR = 12 * MONTH;

  const diffInMs = new Date() - new Date(value);
  
  // 1시간 이내
  const diffInMinutes = Math.floor(diffInMs / MINUTE);
  const diffInHours = Math.floor(diffInMs / HOUR);
  if (diffInHours > 1) {
    return diffInMinutes === 1 ? '1 minute ago' : `${diffInMinutes} minutes ago`;
  }

  // 하루 이내
  const diffInDays = Math.floor(diffInMs / DAY);
  if (diffInDays > 1) {
    return diffInHours === 1 ? '1 hour ago' : `${diffInHours} hours ago`;
  }

  // 1달 이내
  const diffInMonths = Math.floor(diffInMs / MONTH);
  if (diffInMonths > 1) {
    return diffInDays === 1 ? '1 day ago' : `${diffInDays} days ago`;
  }

  // 1년 이내
   const diffInYears = Math.floor(diffInMs / YEAR);
   if (diffInYeares > 1) {
     return diffInMonths === 1 ? '1 month ago' : `${diffInMonths} months ago`;
   }
    return diffInYears === 1 ? '1 year ago' : `${diffInYears} years ago`;
}
```